validation/implicit_free_surface/geostrophic_adjustment_test.jl

```diff
@@ -0,0 +1,98 @@
+using Oceananigans
+using Oceananigans.Units
+using Oceananigans.Models.HydrostaticFreeSurfaceModels: ImplicitFreeSurface
+using Statistics
+
+function geostrophic_adjustment_simulation(free_surface)
+
+    grid = RegularRectilinearGrid(size = (64, 1, 1),
+                                  x = (0, 1000kilometers), y = (0, 1), z = (-400meters, 0),
+                                  topology = (Bounded, Periodic, Bounded))
```

Comment:
Out of curiosity, it should work if we picked the grid to be Flat in y, right?

Reply:
Yes it should! Actually, this was crux of much of my pain in developing this feature. I was missing a fill_halo_regions and as a result the Periodic case kept failing. I think the example uses Flat. We could use it here too, but periodic might be better for catching bugs (I guess ideally we'd test all topologies...)

Reply:
Sounds good to me.  Nothing needs to be changed I don't think.